### PR TITLE
Dogs query w distance

### LINKED ISSUE #67: Implement distance as a GraphQL attribute on users
The logic for figuring out distance between users is complete, it just needs to be added to the GraphQL schema.

### PR DESCRIPTION
**Pull request by chakeresa**
**Changes proposed in this pull request:**
 - Move `::sorted_by_distance` from `User` to `Dog`
 - Add `distance` attribute to `UserType` and `DogType`
 - Update `dogs` query to use `Dog::sorted_by_distance` if authenticated `current_user` has a  location. In that case, dogs whose owner does not have a location are not included in the results.
 - Order `users` and unauthenticated `dogs` queries by `id` to prevent intermittent test failures
 - DRY up `CurrentUserType` by inheriting from `Types::UserType`
 - Mock time for tests of age queries
 - Raise error in `Location before_save :update_latlong` if the address input is invalid (no geocoding results)
 - Catch edge cases in `User#distance_to` for the current user or the target dog/user not having a location -- return nil
 - Add actual addresses for location seeds

Resolves #67, related to user story #7 

**The following checks have been completed:**
 - [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
 - [x] Checked `coverage/index.html` - did not add any new code that's not covered by testing (if possible)
 - [x] Merged in the latest master to my branch with `git pull origin master` & resolved merge conflicts
 - [x] Ran `rails db:migrate`
 - [x] Ran the test suite - all tests are passing (or maybe skipped)
 - [x] Checked affected endpoints in Postman / GraphiQL
 - [x] Updated README for changes (new endpoints, new gems, etc)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ https://crowdhound-be.herokuapp.com/
```
Comment (alexander-mathieu) [inline README.md]: 👍

File: spec/graphql/queries/current_user_spec.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def query
     <<~GQL
```
Comment (alexander-mathieu) [inline spec/graphql/queries/current_user_spec.rb]: Good catch.

File: db/seeds.rb [hunk 3]
```diff
@@ -26,8 +26,32 @@
 create_list(:photo, 2, photoable: dog22)
```
Comment (alexander-mathieu) [inline db/seeds.rb]: Yay, real locations!

File: spec/models/dog_spec.rb [hunk 4]
```diff
@@ -21,13 +21,69 @@
 
   describe 'instance methods' do
     before :each do
-      allow(Time).to receive(:now).and_return('Fri, 15 Apr 2018'.to_time)
```
Comment (alexander-mathieu) [inline spec/models/dog_spec.rb]: 🤦‍♂
